### ISSUE TITLE
Can't be deployed to Deno Deploy

### ISSUE THREAD

**Issue by doga** [sergetymo/kaksik#4]
Congrats, kaksik seems to work just fine for self-hosted Gemini servers.

But I haven't had any luck deploying to [Deno Deploy](https://deploy.deno.com/).

Here is what I get when deploying to Deno Deploy:

```text
  deployctl deploy
  ℹ Using config file '.../deno.json'
  ✔ Deploying to project gemini-docs.
  ✔ Entrypoint: .../main.ts
  ℹ Uploading all files from the current dir (...)
  ✔ Found 14 assets.
  ✔ Uploaded 4 new assets.
  ✖ Deployment failed.
  error: The deployment failed: UNCAUGHT_EXCEPTION

  TypeError: Deno.listenTls is not a function
      at Application.start (https://deno.land/x/kaksik@v1.2.1/classes/Application.ts:32:24)
      at file:///src/main.ts:17:11
```

Here is `main.ts`:

```js
import { Application } from './deps.mts';

const
isTest = Deno.env.get('THIS_IS_A_TEST'),
domain = isTest ? 'localhost' : '<my DNS address>',
app    = new Application({
  keyFile : `./cert/${domain}/key.pem`,
  certFile: `./cert/${domain}/cert.pem`,
})

app.use(ctx => {
  ctx.response.body = '# Hello World!'
})

await app.start()
```

Before deploying, I have uploaded my own RSA certificate and private key to the project, and Deno Deploy has accepted them. `keyFile` and `certFile` shouldn't be needed for Deno Deploy.

**Comment by sergetymo** [comment]
Hello @doga!

Thanks for filing the issue. Kaksik was developed quite a while ago and is indeed used probably only by me for a couple self-hosted Gemini apps and static resources. I've never tested it against services like Deno Deploy you mentioned.

Also, judging by the brief look on an error you provided, Deno API evolved quite a lot since Kaksik was developed, therefore there might be necessity to update the library.

Looking at above, I believe my plan would be
- [ ] Update Deno binary to latest and test Kaksik with it.
- [ ] Figure out nuances of its usage within Deno Deploy and similar services, especially in regards of using encryption certificates.
- [ ] Update library code to conform to the environment said services are providing.

Please don't expect prompt updates though, Kaksik is just a pet project.

Thanks!

**Comment by doga** [comment]
Hello @sergetymo ,

Thanks.

I've filed this issue because Gemini clients are using TLS anyway, so I think it's safe for server libraries to allow both TLS servers and "cleartext" servers, in case the hosting service takes care of TLS.

That's why [I've forked Kaksik](https://github.com/doga/gemini-server) to fix Application.ts to allow both modes. It seems Deno Deploy can't serve Gemini anyway, I think they are only hosting web servers. But this fix may be useful for other hosting services.

I think Gemini is much more suitable for browsing from the terminal than the WWW. The `w3m` browser is a nightmare to use for instance.

I first became aware of Gemini yesterday, and my initial thought was, Gemini is ideal for browsing software docs (JSdoc etc) from the terminal. That's why I've spent a couple hours on this yesterday and today.

Great work!
